rc/base/file.kak

```diff
@@ -0,0 +1,13 @@
+decl str mimetype
+
+hook global BufOpen .* %{ %sh{
+    if [ -z "${kak_opt_filetype}" ]; then
+        mime=$(file -b --mime-type "${kak_buffile}")
+        printf %s\\n "${mime}" | grep -q '^text/x-'
```

Comment:
Looking at the other changes that were needed, I suggest we use a mapping here in the end, something like:
```
case $mime in
  text/x-c++) filetype=cpp ;;
  text/x-tex) filetype=latex ;;
  text/x-shellscript) filetype=sh ;;
  text/x-*) filetype=${mime%text/x-} ;;
  text/plain) ;;
  text/*) filetype=${mime%text/} ;;
esac
[ -n "$filetype" ] && printf "set buffer filetype '%s'\n" ${filetype}
```